scipy/optimize/nonlin.py

```diff
@@ -272,7 +272,7 @@ def nonlin_solve(F, x0, jacobian='krylov', iter=None, verbose=False,
     func = lambda z: _as_inexact(F(_array_like(z, x0))).flatten()
     x = x0.flatten()
 
-    dx = np.inf
+    dx = np.full_like(x, np.inf)
```

Comment:
Would it make sense to add a regression test based on the nice example you provided in the matching issue?

I'm not super familiar with this functionality so an `optimize` regular may want to check as well.

Reply:
Passing a locally defined `tol_norm` is already tested

https://github.com/scipy/scipy/blob/b9945be833ad0c01410d26faab5c69a40371a8b8/scipy/optimize/tests/test_nonlin.py#L142-L144

however this doesn't fail when it's called on `dx=np.inf` since it uses `np.absolute` which turns a `float` which doesn't have `.max` into a `numpy.float64` which does.  If it's changed to
```python
return abs(x).max()
```
then 
```shell
 pytest -k test_tol_norm_called scipy/optimize/tests/test_nonlin.py 
```
in current master raises
> E       AttributeError: 'float' object has no attribute 'max'
but passes with this PR.

Reply:
I'm not sure what the original intention of using `np.absolute` rather than `abs` was, but I've committed this change since it passes all the other tests and does reveal #12813.  Thanks!